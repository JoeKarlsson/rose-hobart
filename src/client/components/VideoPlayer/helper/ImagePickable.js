/*
  eslint
	react/forbid-prop-types: 0
*/

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class ImagePickable extends PureComponent {
	constructor() {
		super();
		this.onClick = this.onClick.bind(this);
	}

	onClick() {
		this.props.onChange(this.props.src);
	}

	render() {
		const { src, selected, style } = this.props;
		return (
			<button
				onClick={this.onClick}
				onKeyPress={this.onClick}
				tabIndex="0"
			>
				<img
					alt="Blur"
					src={src}
					style={{
						borderWidth: 2,
						borderColor: selected ? '#F00' : 'transparent',
						borderStyle: 'solid',
						...style,
					}}
				/>
			</button>
		);
	}
}

ImagePickable.propTypes = {
	style: PropTypes.any,
	src: PropTypes.string,
	selected: PropTypes.bool,
	onChange: PropTypes.func,
};

ImagePickable.defaultProps = {
	style: '',
	src: '',
	selected: false,
	onChange: () => '',
};

export default ImagePickable;
