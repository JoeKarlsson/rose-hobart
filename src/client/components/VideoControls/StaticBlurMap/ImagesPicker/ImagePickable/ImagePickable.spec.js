import React from 'react';
import renderer from 'react-test-renderer';
import {
	shallow,
	configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ImagePickable from './ImagePickable';

configure({ adapter: new Adapter() });

describe('ImagePickable', () => {
	let wrapper;
	let inst;
	const mockhandleChange = jest.fn();

	beforeEach(() => {
		wrapper = shallow(<ImagePickable onChange={mockhandleChange} />);
		inst = wrapper.instance();
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('match the snapshot', () => {
				const component = renderer.create(<ImagePickable onChange={mockhandleChange} />);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});
			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});
			it('should not have any inital props', () => {
				const initialProps = inst.props;
				const expectedProps = {
					style: '',
					src: '',
					selected: false,
				};
				expect(initialProps).toMatchObject(expectedProps);
			});
			it('should not have any inital state', () => {
				const initialState = inst.state;
				const expectedState = {};
				expect(initialState).toMatchObject(expectedState);
			});
		});
	});
});
