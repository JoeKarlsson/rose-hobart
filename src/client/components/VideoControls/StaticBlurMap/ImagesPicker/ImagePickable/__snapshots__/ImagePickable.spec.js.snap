// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePickable rendering initial state match the snapshot 1`] = `
<button
  onClick={[Function]}
  onKeyPress={[Function]}
  tabIndex="0"
>
  <img
    alt="Blur"
    src=""
    style={
      Object {
        "borderColor": "transparent",
        "borderStyle": "solid",
        "borderWidth": 2,
      }
    }
  />
</button>
`;
